src/framework/components/light/component.js

```diff
@@ -1224,12 +1242,20 @@ class LightComponent extends Component {
     }
 
     onEnable() {
+        const scene = this.system.app.scene;
+        const layers = scene.layers;
+
         this.light.enabled = true;
 
-        this.system.app.scene.on('set:layers', this.onLayersChanged, this);
-        if (this.system.app.scene.layers) {
-            this.system.app.scene.layers.on('add', this.onLayerAdded, this);
-            this.system.app.scene.layers.on('remove', this.onLayerRemoved, this);
+        this._evtLayersChanged?.off();
```

Comment:
why do you call `off` on all these here? They should be always null here, unless there is an engine bug right?
In which case, maybe `Debug.assert(this._evtLayersChanged == null);` should be better?

Reply:
This code, does not change behavior from old one.

There might a bug indeed. onEnable can be called for RenderComponent multiple times, there is no check if it is enabled or not.
So this code does not change the behavior, as before that fix, removes would happen also, regardless if there was onEnable called, even multiple times.
Fixing a multi-calls, should be a separate PR.

Reply:
You are right, off was not called here. Do you want me to remove this line then?

Reply:
yep please. Keen to do an engine release later today, if you have a chance to remove those soonish it'd be awesome.

Reply:
note that this needs to be done in all files I think, off was not called in any of them.